### PR TITLE
Pid render limits

### PR DESCRIPTION
**Pull request by howard-stearns**
When under load, the PID controller determines a sphere around the camera such that objects that (approximately) penetrate that sphere can be rendered, while maintaining half the desired frame rate. Entities and avatars lying entirely outside the sphere are not rendered.  There is a preference for the smallest such sphere, defaulting to 50m. (That's the only "knob".)

There is a prefence checkbox to instead use the older "acuity" mechanism. When not selected, the various preferences, tools, menus and reports for this mechanism are disabled.

The separate avatar PID controller is removed (which means avatars are always rendered when using acuity).

### CODE REVIEW

**Comment by howard-stearns** [comment]
The work here is "complete", but I don't have a stable test environment set up. I'd to have such a space for proper tuning and comparison between the mechanisms.

Quite a bit of the PR is to maintain the two mechanisms, or to switch over from the avatar-specific one.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 75814e712c9ed4804b36d4dcc7919dc371b8307b available at: https://gist.github.com/e2613e29c64b94cc82ea<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6681-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6681.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 75814e712c9ed4804b36d4dcc7919dc371b8307b available at: https://gist.github.com/e2613e29c64b94cc82ea<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6681-win64.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6681.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 75814e712c9ed4804b36d4dcc7919dc371b8307b available at: https://gist.github.com/e2613e29c64b94cc82ea<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6681.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6681.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9a681766931c941b678a31f96168d382e69b5f8c available at: https://gist.github.com/e2613e29c64b94cc82ea<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6681.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6681.dmg

**Comment by howard-stearns** [comment]
test this please

**Comment by howard-stearns** [comment]
A good place to try this is the domain 'howard/toybox' (when the entity server doesn't crash). Usually, the octree works flawlessly, and then within the portion near you, the horizon cuts out at a few hundred meters.

I still want to work some on the relationship between avatar testing and behavior, but I think this is a step forward as is.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 56a0ef146f5cfc6db7d13a999dd8e2a1be82b3d8 available at: https://gist.github.com/e2613e29c64b94cc82ea<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6681.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6681.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 41052083c3fd141747c6ef4f7bd86c9582c0cb4e available at: https://gist.github.com/e2613e29c64b94cc82ea<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6681.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6681.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d0f64b16caea25b61d45c729f834ba6d8041bb6f available at: https://gist.github.com/e2613e29c64b94cc82ea<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6681.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6681.dmg

File: interface/src/LODManager.cpp [hunk 1]
```diff
@@ -217,15 +238,59 @@ QString LODManager::getLODFeedbackText() {
...
+    float distance;
+    if (!isThrottled) {
+        _renderDistanceController.setMeasuredValueSetpoint(targetFps / 2.0f); // No problem updating in flight.
+        // The PID controller raises the controlled value when the measured value goes up.
+        // The measured value is frame rate. When the controlled value (1 / render cutoff distance)
+        // goes up, the render cutoff distance gets closer, the number of rendered avatars is less, and frame rate
+        // goes up.
+        distance = 1.0f / _renderDistanceController.update(measuredFps, deltaTime);
+    }
+    else {
```
Comment (AndrewMeadows) [inline interface/src/LODManager.cpp]: Bad formatting on the **else** clause.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 342d3e80bc87be9d00eb22b4582d4c6df132d9f1 available at: https://gist.github.com/e2613e29c64b94cc82ea<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6681.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6681.dmg

**Pull request merged**